contracts/FIFOMarket.sol

```diff
@@ -532,7 +540,9 @@ contract FIFOMarket is
       _removeActiveSupplier(supplierAddress);
     }
     // todo any checks on whether this id was already in there?
-    require(_reservedSupply.add(removalId), "Market: Removal already reserved"); // TODO (Gas Optimization): Use custom error
+    if (!_reservedSupply.add(removalId)) {
+      revert RemovalAlreadyReserved({removalId: removalId});
```

Comment:
https://github.com/nori-dot-eco/contracts/pull/280/files#r913286493

Reply:
I'm going to leave these errors here since `_reservedSupply` is just a straight up enumerable set and we do care about the return value in this case

Reply:
(and `add` is a method in the open zeppelin enumerable set upgradeable utility, which is why we can't push this even deeper)

Reply:
oh, that's the bit i missed!